transport.go

```diff
@@ -34,13 +34,11 @@ type Transporter interface {
 	// TODO: Use a better name.
 	IsConnected() bool
 
-	Run() error
-	RunAsync() <-chan error
-	RegisterProtocol(p Protocol) error
-}
+	// ReceiveFramesAsync starts processing incoming frames in a
+	// background goroutine, if it's not already happening.
+	ReceiveFramesAsync()
```

Comment:
I don't think we actually need this to return a channel, since clients can just use Done(), right?